### PR TITLE
ns1: add support for NAPTR

### PR DESCRIPTION
**Pull request by costasd**
Enable support for NAPTR support in NS1.

Closes: #1407

### CODE REVIEW

File: providers/ns1/ns1Provider.go [hunk 1]
```diff
@@ -173,6 +175,20 @@ func buildRecord(recs models.Records, domain string, id string) *dns.Record {
 			rec.AddAnswer(&dns.Answer{Rdata: strings.Split(fmt.Sprintf("%v %s %s", r.CaaFlag, r.CaaTag, r.GetTargetField()), " ")})
 		} else if r.Type == "SRV" {
 			rec.AddAnswer(&dns.Answer{Rdata: strings.Split(fmt.Sprintf("%d %d %d %v", r.SrvPriority, r.SrvWeight, r.SrvPort, r.GetTargetField()), " ")})
+		} else if r.Type == "NAPTR" {
+			rec.AddAnswer(&dns.Answer{Rdata: []string{
+				fmt.Sprintf("%d", r.NaptrOrder),
```
Comment (tlimoncelli) [inline providers/ns1/ns1Provider.go:180]: Consider using `strconv.Itoa()` for these.

Comment (costasd) [reply providers/ns1/ns1Provider.go]: Done, thanks!

**Comment by costasd** [comment]
Removed TLSA support, as I opened a new ns1 account for testing and the record isn't supported in that account.

Will readd TLSA in a new PR as soon as I clarify TLSA support in NS1

**Comment by tlimoncelli** [comment]
Two more:

```
$ staticcheck ./... | grep -i ns1
providers/ns1/ns1Provider.go:179:6: the argument is already a string, there's no need to use fmt.Sprintf (S1025)
$ golint ./...
providers/ns1/ns1Provider.go:219:4: don't use underscores in Go names; var x_ans should be xAns
```

**Comment by costasd** [comment]
> Two more:
> 
> ```
> $ staticcheck ./... | grep -i ns1
> providers/ns1/ns1Provider.go:179:6: the argument is already a string, there's no need to use fmt.Sprintf (S1025)
> $ golint ./...
> providers/ns1/ns1Provider.go:219:4: don't use underscores in Go names; var x_ans should be xAns
> ```

Fixed, thanks! I have to configure an ns1 pipeline at some point :)

**Comment by tlimoncelli** [comment]
Thanks for the PR!

**Pull request merged**